e2e/e2e_test.go

```diff
@@ -169,9 +398,90 @@ func (c *ClientInfo) deletePod(pod *core.Pod) error {
 		return err
 	}
 
-	if err := WaitForPodToDisappear(c.Client, pod.GetNamespace(), pod.GetName(), deleteTimeout); err != nil {
+	if err := WaitForPodToDisappear(c.Client, pod.GetNamespace(), pod.GetName(), podDeleteTimeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *ClientInfo) provisionReplicaSet(replicaCount int32, label, annotations map[string]string) (*apps.ReplicaSet, error) {
+	replicaSet, err := c.Client.AppsV1().ReplicaSets(testNamespace).Create(
+		context.Background(),
+		replicaSetObject(replicaCount, label, annotations),
+		metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if err := WaitForPodBySelector(c.Client, testNamespace, wbLabelEqual, rsCreateTimeout); err != nil {
+		return nil, err
+	}
+
+	replicaSet, err = c.Client.AppsV1().ReplicaSets(testNamespace).Get(context.Background(), replicaSet.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return replicaSet, nil
+}
+
+func (c *ClientInfo) updateReplicaSet(replicaSet *apps.ReplicaSet) (*apps.ReplicaSet, error) {
+	replicaSet, err := c.Client.AppsV1().ReplicaSets(testNamespace).Update(context.Background(), replicaSet, metav1.UpdateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return replicaSet, nil
+}
+
+func (c *ClientInfo) deleteReplicaSet(replicaSet *apps.ReplicaSet) error {
+	const rsDeleteTimeout = 2 * rsCreateTimeout
+	if err := c.Client.AppsV1().ReplicaSets(testNamespace).Delete(context.Background(), replicaSet.Name, metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+
+	if err := WaitForReplicaSetToDisappear(c.Client, replicaSet.GetNamespace(), replicaSet.GetName(), rsDeleteTimeout); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v envVars) maxReplicas(allPods []core.Pod) int32 {
+	const maxPodsPerNode = 110
+	return int32(
+		(v.numComputeNodes*maxPodsPerNode - (len(allPods))) * v.fillPercentCapacity / 100)
+}
+
+// Waits for all replicas to be fully removed from replicaset, and checks that there are 0 ip pool allocations
+func checkZeroIPPoolAllocationsAndReplicas(clientInfo *ClientInfo, k8sIPAM *kubeClient.KubernetesIPAM) error {
+	const (
+		emptyReplicaSet = 0
+		rsSteadyTimeout = 2 * rsCreateTimeout
+	)
+	var err error
+
+	// Set to 0 replicas
```

Comment:
nit: ditch this comment; by using below the `emptyReplicaSet` const, it's pretty obvious what you're doing.